_health-care/_js/components/HealthCareApp.jsx

```diff
@@ -52,6 +52,7 @@ class HealthCareApp extends React.Component {
     this.context.store.dispatch(ensureFieldsInitialized(path));
     if (validations.isValidSection(path, sectionData)) {
       hashHistory.push(this.getUrl('next'));
+      sectionData.sectionComplete = true;
```

Comment:
@akainic is correct.

FYI, this is also why I want to move to Immutable.js.  When we get to that world, such code woudl produce no change in the UI, versus right now you're going to get unreliable updates (it'll change sometimes, but not other times).